pkg/file/redundancy/getter/getter.go

```diff
@@ -97,10 +100,39 @@ func (g *decoder) Get(ctx context.Context, addr swarm.Address) (swarm.Chunk, err
 	}
 	select {
 	case <-g.waits[i]:
-		return swarm.NewChunk(addr, g.rsbuf[i]), nil
 	case <-ctx.Done():
-		return nil, ctx.Err()
+		select {
+		case <-g.ready:
+			select {
+			case <-g.waits[i]:
+			case <-time.After(1 * time.Second):
+				return nil, ctx.Err()
+			}
+		default:
+			return nil, ctx.Err()
+		}
+	}
+	return swarm.NewChunk(addr, g.getData(i)), nil
+}
+
+// setData sets the data shard in the RS buffer
+func (g *decoder) setData(i int, chdata []byte) {
+	data := chdata
+	// pad the chunk with zeros if it is smaller than swarm.ChunkSize
+	if len(data) < swarm.ChunkWithSpanSize {
+		g.lastLen = len(data)
+		data = make([]byte, swarm.ChunkWithSpanSize)
+		copy(data, chdata)
 	}
+	g.rsbuf[i] = data
+}
+
+// getData returns the data shard from the RS buffer
+func (g *decoder) getData(i int) []byte {
+	if i == g.shardCnt-1 && g.lastLen > 0 {
+		return g.rsbuf[i][:g.lastLen] // cut padding
+	}
```

Comment:
why calculate it when it is known from the retrieved chunk. 
we need to remember the length after. retrieval so that when it is padded to full for RS decoding (assumes uniform shard length)